packages/amplify_auth_cognito/ios/Classes/AuthErrorHandler.swift

```diff
@@ -0,0 +1,147 @@
+    }
+    if case .notAuthorized(let localizedError,  let recoverySuggestion, let error) = error {
+        let errorCode = error != nil ? "\(error!)" : "notAuthorized"
+        SwiftAuthCognito.logErrorContents(messages: [localizedError, recoverySuggestion, errorCode])
+        formatError(flutterResult: flutterResult, errorCode: errorCode, msg: msg, localizedError: localizedError, recoverySuggestion: recoverySuggestion)
+    }
+    if case .validation(let field, let localizedError, let recoverySuggestion, let error) = error {
+        let errorCode = error != nil ? "\(error!)" : "validation"
+        SwiftAuthCognito.logErrorContents(messages: [field, localizedError, recoverySuggestion, errorCode])
+        formatError(flutterResult: flutterResult, errorCode: errorCode, msg: msg, localizedError: localizedError, recoverySuggestion: recoverySuggestion)
+    }
+    if case .signedOut(let localizedError, let recoverySuggestion, let error) = error {
+        let errorCode = error != nil ? "\(error!)" : "signedOut"
+        SwiftAuthCognito.logErrorContents(messages: [localizedError, recoverySuggestion, errorCode])
+        formatError(flutterResult: flutterResult, errorCode: errorCode, msg: msg, localizedError: localizedError, recoverySuggestion: recoverySuggestion)
+    }
+    if case .sessionExpired(let localizedError, let recoverySuggestion, let error) = error {
+        let errorCode = error != nil ? "\(error!)" : "sessionExpired"
+        SwiftAuthCognito.logErrorContents(messages: [localizedError, recoverySuggestion, errorCode])
+        formatError(flutterResult: flutterResult, errorCode: errorCode, msg: msg, localizedError: localizedError, recoverySuggestion: recoverySuggestion)
+    }
+}
+
+func formatError(flutterResult: FlutterResult, errorCode: String, msg: String, localizedError: String, recoverySuggestion: String) {
+    var errorMap: [String: Any] = formatErrorMap(errorCode: errorCode, localizedError: localizedError)
+    errorMap["PLATFORM_EXCEPTIONS"] = platformExceptions(localizedError: localizedError, recoverySuggestion: recoverySuggestion)
+    prepareError(flutterResult: flutterResult,  msg: msg, errorMap: errorMap)
+}
+
+
+func prepareError(flutterResult: FlutterResult, msg: String, errorMap: [String: Any]) {
+    flutterResult(FlutterError(
+                    code: "AmplifyException",
+                    message: msg,
+                    details: errorMap)
+    )
+}
+
+func platformExceptions(localizedError: String, recoverySuggestion: String) -> [String: String] {
+    var platformDict: [String: String] = [:]
+    platformDict["platform"] = "iOS"
+    platformDict["localizedErrorMessage"] = localizedError
+    platformDict["recoverySuggestion"] = recoverySuggestion
+    return platformDict
+}
+
+func formatErrorMap(errorCode: String, localizedError: String = "") -> [String: Any] {
+    var errorDict: [String: Any] = [:]
+    switch errorCode {
```

Comment:
Is `switch the right construct here? Can you just convert the camel-case to screaming-snake and have only one statement to assign the `errorDict` value?

Alternately, you could form a map from e.g., `"codeExpired"` to `"CODE_EXPIRED"`, and then do a single assignment.

Reply:
@jamesonwilliams Initially I did write this as a case conversion, but found that there are some inconsistencies, such as:
```
    case "softwareTokenMFANotEnabled":
        errorDict["SOFTWARE_TOKEN_MFA_NOT_FOUND"] = localizedError
```

I think I will implement your map suggestion.  It saves some lines.